pkg/operator/controller/ingress/controller.go

```diff
@@ -501,31 +501,40 @@ func filterTLS13Ciphers(ciphers []string) []string {
 // the finalizer.
 func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController) error {
 	errs := []error{}
-	if err := r.finalizeLoadBalancerService(ingress); err != nil {
-		errs = append(errs, fmt.Errorf("failed to finalize load balancer service for %s/%s: %v", ingress.Namespace, ingress.Name, err))
+	svc, err := r.finalizeLoadBalancerService(ingress)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("failed to finalize load balancer service for ingress %s/%s: %v", ingress.Namespace, ingress.Name, err))
+	}
+	if svc != nil {
```

Comment:
Better to use `else if`.  It should not matter in this case since `finalizeLoadBalancerService` returns a nil service value when it returns a non-nil error value, but generally non-error return values should be ignored in the case of a non-nil error value.